docs/guides/typescript.md

```diff
@@ -65,12 +63,188 @@ const lightMachine = createMachine<LightContext, LightEvent>({
 });
 ```
 
-Providing the context and events as generic parameters for the `createMachine()` function gives many advantages:
+Providing the context and events to the `schema` attribute gives many advantages:
 
 - The context type/interface (`TContext`) is passed on to actions, guards, services and more. It is also passed to deeply nested states.
 - The event type (`TEvent`) ensures that only specified events (and built-in XState-specific ones) are used in transition configs. The provided event object shapes are also passed on to actions, guards, and services.
 - Events which you send to the machine will be strongly typed, offering you much more confidence in the payload shapes you'll be receiving.
 
+## The VSCode Extension <Badge text="4.29+" />
+
+Using our [VS Code extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode), you can automatically generate intelligent typings for XState.
+
+Here's how you can get started:
+
+1. Download and install the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=mattpocock.xstate-vscode).
+2. Open a new file and create a new machine, passing the schema attributes:
+
+```ts
+import { createMachine, t } from 'xstate';
+
+const machine = createMachine({
+  schema: {
+    context: t<{ value: string }>(),
+    events: t<{ type: 'FOO'; value: string } | { type: 'BAR' }>()
+  },
+  initial: 'a',
+  states: {
+    a: {
+      on: {
+        FOO: {
+          actions: 'consoleLogValue',
+          target: 'b'
+        }
+      }
+    },
+    b: {
+      entry: 'consoleLogValueAgain'
+    }
+  }
+});
+```
+
+3. Add `tsTypes: true` to the machine and save the file:
```

Comment:
Perhaps this should use something that is more compatible with the injected replacement
```suggestion
3. Add `tsTypes: {}` to the machine and save the file:
```
```suggestion
3. Add `tsTypes: {}` to the machine and save the file:
```